pootle/apps/pootle_app/management/commands/calculate_checks.py

```diff
@@ -14,69 +14,95 @@
 # This must be run before importing Django.
 os.environ['DJANGO_SETTINGS_MODULE'] = 'pootle.settings'
 
-from django.conf import settings
-from django.core.urlresolvers import set_script_prefix
-from django.utils.encoding import force_unicode
+from django.utils import timezone
 
-from django_rq import job
-
-from pootle.core.cache import get_cache
 from pootle.core.mixins.treeitem import CachedMethods
-from pootle_store.models import Store
-
-from .refresh_stats import Command as RefreshStatsCommand
+from pootle_store.models import Store, QualityCheck, Unit
+from pootle_store.util import OBSOLETE
 
+from . import PootleCommand
 
-logger = logging.getLogger('stats')
-cache = get_cache('stats')
 
-
-class Command(RefreshStatsCommand):
-    help = "Allow stats and text indices to be refreshed manually."
+class Command(PootleCommand):
+    help = "Allow checks to be recalculated manually."
 
     shared_option_list = (
         make_option('--check', action='append', dest='check_names',
                     help='Check to recalculate'),
     )
-    cached_methods = [CachedMethods.CHECKS]
 
-    def handle_noargs(self, **options):
-        calculate_checks.delay(**options)
-        option_list = map(lambda x: '%s=%s' % (x, options[x]),
-                          filter(lambda x: options[x], options))
-        self.stdout.write('calculate checks RQ job added with options: %s. %s.' %
-                          (', '.join(option_list),
-                          'Please make sure rqworker is running'))
+    cached_methods = [CachedMethods.CHECKS]
+    process_disabled_projects = True
+
+    def handle_all_stores(self, translation_project, **options):
+        store_fk_filter = {
```

Comment:
This doesn't look like the right place for defining the filter, but rather in `calculate_checks()`. Just pass the `translation_project` around.